### PR TITLE
YLW-3115: Create and use inheritance

### PR DESCRIPTION
**Pull request by davidgreenland**
#### What

Expands the application by creating derived classes of Student and Teacher extending the base class Person.

#### Why

To learn about inheritance in C# and understand how to override methods in a derived class.

#### How

- Used virtual and override key words to create unique Greet methods for each Type derived from Person. 
- The Student class is extended further with a Ginger class. 
- The HairColour variable is set to null in the base class and overridden in the Ginger constructor. 
- Teacher class also has unique Greet method and a Department property.
- Students have a SubjectChoices property which could contain a list of Lessons or Subject class in the future.

### CODE REVIEW

File: LearningCSharp/Ginger.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+﻿class Ginger : Student
```
Comment (SiddiqueAbdullah) [inline LearningCSharp/Ginger.cs]: make scope clear by adding public/internal/private

File: LearningCSharp/Student.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
+﻿using LearningCSharp;
+
+class Student : Person
+{
+    public int YearGroup { get; protected set; }
+    public List<string> SubjectChoices { get; set; } = new List<string>();
+
```
Comment (SiddiqueAbdullah) [inline LearningCSharp/Student.cs:7]: where is the grade field?

**Review by SiddiqueAbdullah — CHANGES_REQUESTED** [review]

**Comment by SiddiqueAbdullah** [comment]
There are also some conflicts, resolve them

File: LearningCSharp/Ginger.cs (hunk 1 above)
Comment (davidgreenland) [reply LearningCSharp/Ginger.cs]: done.

File: LearningCSharp/Student.cs (hunk 2 above)
Comment (davidgreenland) [reply LearningCSharp/Student.cs:7]: Do you mean like "my child is in 5th grade"? I thought I'd use YearGroup as I'm more familiar with that.

**Review by SiddiqueAbdullah — APPROVED** [review]

**Pull request merged**